datasets/caner/caner.py

```diff
@@ -0,0 +1,133 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A new corpus of tagged data that can be useful for handling the issues in recognition of Classical Arabic named entities"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import datasets
+
+
+_CITATION = """\
+@article{article,
+author = {Salah, Ramzi and Zakaria, Lailatul},
+year = {2018},
+month = {12},
+pages = {},
+title = {BUILDING THE CLASSICAL ARABIC NAMED ENTITY RECOGNITION CORPUS (CANERCORPUS)},
+volume = {96},
+journal = {Journal of Theoretical and Applied Information Technology}
+}
+"""
+
+_DESCRIPTION = """\
+Classical Arabic Named Entity Recognition corpus as a new corpus of tagged data that can be useful for handling the issues in recognition of Arabic named entities.
+"""
+
+_HOMEPAGE = "https://github.com/RamziSalah/Classical-Arabic-Named-Entity-Recognition-Corpus"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_URL = "https://github.com/RamziSalah/Classical-Arabic-Named-Entity-Recognition-Corpus/archive/master.zip"
+
+
+class Caner(datasets.GeneratorBasedBuilder):
+    """Classical Arabic Named Entity Recognition corpus as a new corpus of tagged data that can be useful for handling the issues in recognition of Arabic named entities"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="first_domain", version=VERSION, description="This covers the training dataset"),
+    ]
+
+    DEFAULT_CONFIG_NAME = "first_domain"
```

Comment:
```suggestion
```
You don't need all this if there's only one config

After this change you'll need to replace all the occurrences of "first_domain" by "default" in the dataset_infos.json file. You'll also need to move the dummy_data.zip file from `datasets/caner/dummy/first_domain/1.1.0` to `datasets/caner/dummy/default/1.1.0`.

Reply:
Thanks for the comments, very helpful. I put the dummy_data zip in `datasets/caner/dummy/default/1.1.0` however the dummy_data tests failed. Putting it in `datasets/caner/dummy/1.1.0` passed the tests. Not sure if this behaviour is unexpected?

Reply:
Thanks !
Yes indeed `datasets/caner/dummy/1.1.0` is the right path, my bad ;)